main.cpp

```diff
@@ -51,23 +51,411 @@ If you need inspiration for what to write, take a look at previously approved st
+
+    struct Vehicle
+    {
+        // 5 member variable
+        int numberOfTires = 4;
+        int maxSpeed;
+        bool turboCharged;
+        std::string vin;
+        float engineDisplacement;
+
+        // 3 member functions
+        void renovateVehicle(std::string vin, int maxSpeed = 150, bool turboCharged = false);
+        float calculateHpToTheWheel(float engineDisplacement, bool turboCharged = true, int maxSpeed = 200);
+        void resetEngineControlModule(std::string vin, bool vehicleIsImmobolized = true);
+
+        // Constructor
+        Vehicle();
+        //Destructor
+        ~Vehicle();
+    };
+};
+
+Car::Car():
+make("Ford"),
+totalMilesOnVehicle(13000),
+sellingPrice(10000.0f),
+fuelType(0)
+{
+    std::cout << "Car constuctor called" << std::endl;
+}
+
+Car::~Car()
+{
+    std::cout << "Car class destructor called" << std::endl;
+}
+
+float Car::calculateTrueSellValue(std::string make, float sellingPrice, int totalMilesOnVehicle)
+{
+    float trueSellValue = 0.0f;
+    
+    std::cout << "Calculating True Sell Value for " << make << std::endl;
+    trueSellValue = (sellingPrice - (.1f * (static_cast<float>(totalMilesOnVehicle))));
+    return trueSellValue;
+}
+
+int Car::calculateMilesUntilOilChange(int fuelType, int totalMilesOnVehicle)
+{
+    int milesUntilOilChange = 0;
+
+    if ((milesUntilOilChange = totalMilesOnVehicle % 5000))
+        std::cout << fuelType << " car needs doesn't need oil change" << std::endl;
+
+    std::cout << "Car needs oil change in " << milesUntilOilChange << " miles"<< std::endl;
+    return milesUntilOilChange;                                       
+}
+    
+bool Car::predictRaceWinProbability(std::string make, int maxSpeed)
+{
+    int competitorMaxSpeed = 200;
+
+    if (maxSpeed > competitorMaxSpeed)
+    {
+        std::cout << "You will most likely win in a race" << std::endl;
+        return true;
+    }
+
+    std::cout << "You will most likely lose in a race" << std::endl;
+    return false;
+}
+
+Car::Vehicle::Vehicle():
+maxSpeed(160),
+turboCharged(false),
+vin("1BBHHK7772122340"),
+engineDisplacement(5.7f)
+{
+    std::cout << "Vehicle constructor called" << std::endl;
+}
+
+Car::Vehicle::~Vehicle()
+{
+    std::cout << "Vehicle Destructor called" << std::endl;
+}
+
+void Car::Vehicle::renovateVehicle(std::string vin, int maxSpeed, bool turboCharged)
+{
+    if((vin[9] < 'J') && (maxSpeed < 155))
+    {
+        std::cout << "Renovate vehicle" << std::endl;
+    }
+
+    std::cout << "Don't renovate vehicle" << std::endl;
+}
+
+float Car::Vehicle::calculateHpToTheWheel(float engineDisplacement, bool turboCharged, int maxSpeed)
+{
+    if(turboCharged)
+    {
+        return((engineDisplacement * 100.0f) + 100.0f + maxSpeed);
+    }
+
+    return ((engineDisplacement * 100.0f) + maxSpeed);
+}
+
+void Car::Vehicle::resetEngineControlModule(std::string vin, bool vehicleIsImmobolized)
+{
+    if((vin[9] > 'J') && vehicleIsImmobolized)
+    {
```

Comment:
you should check if vin's length is `> 9+1` (because indexing starts at `0`, and `[9]` is the 10th element) before you check the value at index `[9]`
```
if( vin.size() >= 10 && vin[9] > 'J' && vehicleIsImmobilized )
```
https://en.cppreference.com/w/cpp/string/basic_string/size
- [ ] revise